src/Adapter/CartRule/CommandHandler/DeleteCartRuleHandler.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\CartRule\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\CartRule\AbstractCartRuleHandler;
+use PrestaShop\PrestaShop\Core\Domain\CartRule\Command\DeleteCartRuleCommand;
+use PrestaShop\PrestaShop\Core\Domain\CartRule\CommandHandler\DeleteCartRuleHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\CartRule\Exception\CannotDeleteCartRuleException;
+
+/**
+ * Handles deletion of cart rule using legacy object model
+ */
+final class DeleteCartRuleHandler extends AbstractCartRuleHandler implements DeleteCartRuleHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(DeleteCartRuleCommand $command): void
+    {
+        $cartRuleId = $command->getCartRuleId();
+        $specificPriceRule = $this->getCartRule($cartRuleId);
```

Comment:
No, it's intended 😄you call them "specific price rule" althought we use `getCartRule()` to retrieve them it seems confusing. Am I missing something here ?

Reply:
it should be a mistake.
SpecificPriceRule was a synonym for CatalogPriceRule used in legacy, but CartRule should stay CartRule :smile:

Reply:
Hmm, I remember I had a reason when I called it that way, but to be honest I can't remember at this point.

I must have mistaken it for in some way CatalogPriceRule. Will fix it.